steamtinkerlaunch

```diff
@@ -7427,43 +7432,59 @@ function installd3d47dll {
 	fi
 }
 
+function createDLReShadeList {
+	RESHADEVERSIONS="$RSVERSOVRD!$RSVERS!5.0.0!4.9.1!3.4.1" # List some hardcoded reshade versions to selct for the override 
+	#TODO replace with a auto poplating list based on the last few releses
+}
+
 function dlReShade {
 	if [ -z "$1" ]; then
 		DLVERS="$RSVERS"
 	else
 		DLVERS="$1"
 	fi
-	rm "$RESHADESRCDIR/$RS_64" 2>/dev/null
-	rm "$RESHADESRCDIR/$RS_32" 2>/dev/null
-	rm "$RESHADESRCDIR/${RS_64//.dll/.json}" 2>/dev/null
-	rm "$RESHADESRCDIR/${RS_32//.dll/.json}" 2>/dev/null
 
-	DLDST="${RESHADESRCDIR}/${RSSU}.exe"
-	DLDSTV="${RESHADESRCDIR}/${RSSU}_${DLVERS}.exe"
+	# No longer needed?
+	#rm "$RESHADESRCDIR/$RS_64" 2>/dev/null
+	#rm "$RESHADESRCDIR/$RS_32" 2>/dev/null
+	#rm "$RESHADESRCDIR/${RS_64//.dll/.json}" 2>/dev/null
+	#rm "$RESHADESRCDIR/${RS_32//.dll/.json}" 2>/dev/null
+
+	#DLDST="${RESHADESRCDIR}/${RSSU}.exe"
+	#DLDSTV="${RESHADESRCDIR}/${RSSU}_${DLVERS}.exe"
+
+	DLDST="${RESHADESRCDIR}/${RSSU}_${DLVERS}.exe"
 
 	RSSETUP="${RESHADEDLURL}/${RSSU}_${DLVERS}.exe"
 
 	dlCheck "$RSSETUP" "$DLDST" "X" "Downloading $RSSU"
-	if [ ! -f "$DLDSTV" ]; then
-		writelog "INFO" "${FUNCNAME[0]} - Creating backup '${RSSU}_${DLVERS}.exe'"
-		cp "$DLDST" "$DLDSTV"
-	fi
-
+	
 	echo "$DLVERS" > "${DLDST//.exe/.log}"
 
 	if [ ! -s "$DLDST" ]; then
 		writelog "SKIP" "${FUNCNAME[0]} - Downloaded file '$DLDST' is empty - removing"
 		rm "$DLDST" 2>/dev/null
 	else
-		"$UNZIP" -q "$DLDST" -d "$RESHADESRCDIR" 2>/dev/null
+		"$UNZIP" -q "$DLDST" -d "$RESHADESRCDIR/$RSVERS" 2>/dev/null
 		writelog "INFO" "${FUNCNAME[0]} - Downloaded and extracted ${RESH}-v${DLVERS} file '$DLDST'"
 	fi
 }
 
+function overrideReShadeVersion
+{
+	if ! [ "$RSVERSOVRD" -eq "$RSVERS" ]; then
```

Comment:
I think I actually suggested this syntax so sorry if this is wrong, but does this check work? If they're both strings you might have to do a check like `if [[ "$RSVERSOVRD" = "RSVERS" ]]; then` (note the double square brackets -- Bash is fussy about this for some reason...)

If it works though, we can leave it as is :-)

Reply:
i think you are right now that I think about it I never did see that condition pass ...

Reply:
Yeah this logic might need a little bit of tweaking. If the two versions are equal we probably won't want to do `RSVERS="$RSVERSOVRD"`.

Maybe it could look like this?

```bash
# Assumes we have a checkbox like in my comment - No rush, just letting you know the context of how I drafted this snippet :-)
if [ "$RSOVRD" -eq 1 ]; then  # This is meant to be the checkbox var
    if [[ ! "$RSOVRDVERS" = "$RSVERS" ]]; then
        writelog "INFO" "${FUNCNAME[0]} - Overriding global ReShade version '$RSVERS' with '$RSVERSOVRD'"
        RSVERS="$RSOVRDVERS"
    else
        writelog "SKIP" "${FUNCNAME[0]} - ReShade Override version and ReShade global version match - Not overriding"
    fi
else
    writelog "SKIP" "${FUNCNAME[0]} - ReShade override is disabled - Skipping"
fi
```